src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.ts

```diff
@@ -100,6 +109,10 @@ export class RgwBucketFormComponent implements OnInit {
       // Add
       this.rgwBucketService.create(bidCtl.value, ownerCtl.value).subscribe(
         () => {
+          this.notificationService.show(
+            NotificationType.success,
+            this.i18n('Created Object Gateway bucket "{{bid}}".', { bid: bidCtl.value })
+          );
```

Comment:
Could you add some test please?

I know there were no submit tests before so I give you guidance how to easily only test the notification message.

```suggestion
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.spec.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.spec.ts
index ee8895e9ab..20a022138a 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.spec.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.spec.ts
@@ -1,15 +1,18 @@
 import { HttpClientTestingModule } from '@angular/common/http/testing';
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { FormControl, ReactiveFormsModule } from '@angular/forms';
+import { Router } from '@angular/router';
 import { RouterTestingModule } from '@angular/router/testing';
 
 import { ToastModule } from 'ng2-toastr';
-import { of as observableOf } from 'rxjs';
+import { of, of as observableOf } from 'rxjs';
 
 import { configureTestBed, i18nProviders } from '../../../../testing/unit-test-helper';
 import { RgwBucketService } from '../../../shared/api/rgw-bucket.service';
 import { SharedModule } from '../../../shared/shared.module';
 import { RgwBucketFormComponent } from './rgw-bucket-form.component';
+import { NotificationService } from '../../../shared/services/notification.service';
+import { NotificationType } from '../../../shared/enum/notification-type.enum';
 
 describe('RgwBucketFormComponent', () => {
   let component: RgwBucketFormComponent;
@@ -98,4 +101,30 @@ describe('RgwBucketFormComponent', () => {
       }
     });
   });
+
+  describe('submit tests', () => {
+    let notificationService: NotificationService;
+    let bucketService = TestBed.get(RgwBucketService);
+
+    const testSuccessfulNotification = (methodName, notificationText) => {
+      spyOn(bucketService, methodName).and.callFake(() => of([]));
+      component.bucketForm.markAsPristine();
+      component.submit();
+      expect(notificationService.show).toHaveBeenCalledWith(
+        NotificationType.success,
+        notificationText
+      );
+    };
+
+    beforeEach(() => {
+      spyOn(TestBed.get(Router), 'navigate').and.stub();
+      notificationService = TestBed.get(NotificationService);
+      spyOn(notificationService, 'show');
+      bucketService = TestBed.get(RgwBucketService);
+    });
+
+    it('tests notification', () => {
+      testSuccessfulNotification('create', 'Created Object Gateway bucket "".');
+    });
+  });
 });
```

Reply:
Done

Reply:
I used a function on purpose so that it will be easier to test the `update` method too ;) Could you please add the `update` test?

Reply:
> I used a function on purpose so that it will be easier to test the `update` method too ;) Could you please add the `update` test?

Fixed